src/validation-middleware.ts

```diff
@@ -47,19 +47,23 @@ export const validationMiddleware = <H, P, Q, B, F, R>(
 
   return async (ctx: DefaultContext, next: Next) => {
     const validated = await Promise.all([
-      validate(ctx.request.headers, validation.headers, 'headers'),
-      validate(ctx.request.params, validation.params, 'params'),
-      validate(ctx.request.query, validation.query, 'query'),
-      validate(ctx.request.body, validation.body, 'body'),
-      validate(ctx.request.files, validation.files, 'files'),
+      validate<H>(ctx.request.headers, validation.headers, 'headers'),
+      validate<P>(ctx.request.params, validation.params, 'params'),
+      validate<Q>(ctx.request.query, validation.query, 'query'),
+      validate<B>(ctx.request.body, validation.body, 'body'),
+      validate<F>(ctx.request.files, validation.files, 'files'),
     ]);
 
-    const inputErrors = validated.reduce((acc: ZodRouterInvalid, curr) => {
-      if (assertZodValidationError(curr)) {
-        acc[curr.requestParameter] = curr.error;
-      }
-      return acc;
-    }, {});
+    const inputErrors = validated.reduce<ZodRouterInvalid<H, P, Q, B, F>>(
+      (acc, curr) => {
+        if (assertZodValidationError<any>(curr)) {
```

Comment:
why do we need <any> here?

Reply:
Oh, we don't, it's a leftover when I was trying to get Typescript to cooperate, I'll remove it.